src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -913,12 +913,49 @@ def _validate_lock_params_match_lock(
                 name, _resource_name))
 
 
-def get_lock(name, resource_group_name=None, resource_provider_namespace=None,
-             parent_resource_path=None, resource_type=None, resource_name=None):
+def _parse_lock_id(id_arg):
+    """
+    Lock ids look very different from regular resource ids, this function uses a regular expression
+    that parses a lock's id and extracts the following parameters if available:
+    -name: the lock's name; always present in a lock id
+    -resource_group_name: the name of the resource group; present in group/resource level locks
+    -resource_provider_namespace: the resource provider; present in resource level locks
+    -resource_type: the resource type; present in resource level locks
+    -resource_name: the resource name; present in resource level locks
+    -parent_resource_path: the resource's parent path; present in child resources such as subnets
+    """
+    regex = re.compile(
+        '/subscriptions/[^/]*(/resource[gG]roups/(?P<resource_group_name>[^/]*)'
+        '(/providers/(?P<resource_provider_namespace>[^/]*)'
+        '(/(?P<parent_resource_path>.*))?/(?P<resource_type>[^/]*)/(?P<resource_name>[^/]*))?)?'
+        '/providers/Microsoft.Authorization/locks/(?P<name>[^/]*)')
+
+    return regex.match(id_arg).groupdict()
+
+
+def _call_correct_subscription_get(lock_client, *args):
+    if supported_api_version(ResourceType.MGMT_RESOURCE_LOCKS, max_api='2015-01-01'):
+        return lock_client.management_locks.get(*args)
+    return lock_client.management_locks.get_at_subscription_level(*args)
+
+
+def get_lock(name=None, resource_group_name=None, resource_provider_namespace=None,
+             parent_resource_path=None, resource_type=None, resource_name=None, ids=None):
     """
     :param name: The name of the lock.
     :type name: str
     """
+    if ids:
+        kwargs_list = []
+        for id_arg in ids:
+            try:
+                kwargs_list.append(_parse_lock_id(id_arg))
+            except AttributeError:
+                logger.error('az lock show: error: argument --ids: invalid ResourceId value: \'%s\'' % id_arg)
+                return
+        results = [get_lock(**kwargs) for kwargs in kwargs_list]
+        return results[0] if len(results) == 1 else results
```

Comment:
/cc @derekbekoe Don't we have structure take care of multiple ids on core level, which expand a namespace's iterable parameter to run the command multiple times. Can you show @williexu how should that parameter be registered?

Reply:
That parameter cannot be registered for locks because the "lock" itself cannot be given a static label. Without making changes to the infrastructure, handling this in the custom logic is the way to go.